java/flight/src/main/java/org/apache/arrow/flight/FlightService.java

```diff
@@ -191,17 +191,17 @@ public void completed() {
             StreamPipe.wrap(responseObserver, PutResult::toProtocol)).run();
         responseObserver.onCompleted();
       } catch (Exception ex) {
-        logger.error("Failed to process custom put.", ex);
         responseObserver.onError(ex);
-        // The client may have terminated, so the exception here is effectively swallowed.
         // Log the error as well so -something- makes it to the developer.
         logger.error("Exception handling DoPut", ex);
-      }
-      try {
-        fs.close();
-      } catch (Exception e) {
-        logger.error("Exception closing Flight stream", e);
-        throw new RuntimeException(e);
+        throw new RuntimeException(ex);
+      } finally {
```

Comment:
FlightStream is AutoCloseable, so a try-with-resources here would make things cleaner.

Reply:
@lihalite thanks a lot for the comment.
It seems the FlightStream object is created in the method, but closed in a different thread. So try-with-resources may not be used here.

Reply:
@lihalite thanks a lot for the comment.

> Sounds good to me, let's get this merged.

Thanks a lot for your careful review.